MSBLOC.Web/Program.cs

```diff
@@ -19,7 +22,41 @@ public static void Main(string[] args)
 
         public static IWebHost BuildWebHost(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration(new Program().MSBLOCConfigureAppConfiguration)
                 .UseStartup<Startup>()
                 .Build();
+
+        public void MSBLOCConfigureAppConfiguration(WebHostBuilderContext context, IConfigurationBuilder config)
+        {
+            config.AddEnvironmentVariables("MSBLOC_");
+
+            var builtConfig = config.Build();
+
+            var azureKeyVault = builtConfig["Azure:KeyVault"];
+
+            if (!string.IsNullOrEmpty(azureKeyVault))
+            {
+                var keyVaultConfigBuilder = GetConfigurationBuilder();
+
+                var keyVaultClient = GetKeyVaultClient();
+
+                keyVaultConfigBuilder.AddAzureKeyVault($"https://{azureKeyVault}.vault.azure.net/", keyVaultClient, new DefaultKeyVaultSecretManager());
+
+                var keyVaultConfig = keyVaultConfigBuilder.Build();
+
+                config.AddConfiguration(keyVaultConfig);
+            }
```

Comment:
Ok one argument to **NOT** make this an extension method would be to add `else { logger.Warn("Key vault not configured . . .")}`.

I think We can load IConfigurationBuilder into the Startup.cs constructor and call those code there, and I know we can put ILogger there as well.